lib/append-to.js

```diff
@@ -28,141 +8,28 @@ module.exports = function appendTo (file, rootElem, cb) {
-
-    if (VIDEOSTREAM_EXTS.indexOf(extname) >= 0) useVideostream()
-    else useMediaSource()
-
-    function useVideostream () {
-      debug('Use `videostream` package for ' + file.name)
-      createElem()
-      elem.addEventListener('error', fallbackToMediaSource)
-      elem.addEventListener('playing', onPlaying)
-      videostream(file, elem)
-    }
-
-    function useMediaSource () {
-      debug('Use MediaSource API for ' + file.name)
-      createElem()
-      elem.addEventListener('error', fallbackToBlobURL)
-      elem.addEventListener('playing', onPlaying)
-
-      file.createReadStream().pipe(new MediaSourceStream(elem, { extname: extname }))
-      if (currentTime) elem.currentTime = currentTime
-    }
-
-    function useBlobURL () {
-      debug('Use Blob URL for ' + file.name)
-      createElem()
-      elem.addEventListener('error', fatalError)
-      elem.addEventListener('playing', onPlaying)
-      file.getBlobURL(function (err, url) {
-        if (err) return fatalError(err)
-        elem.src = url
-        if (currentTime) elem.currentTime = currentTime
-      })
-    }
-
-    function fallbackToMediaSource (err) {
-      debug('videostream error: fallback to MediaSource API: %o', err.message || err)
-      elem.removeEventListener('error', fallbackToMediaSource)
-      elem.removeEventListener('playing', onPlaying)
-
-      useMediaSource()
-    }
-
-    function fallbackToBlobURL (err) {
-      debug('MediaSource API error: fallback to Blob URL: %o', err.message || err)
-      elem.removeEventListener('error', fallbackToBlobURL)
-      elem.removeEventListener('playing', onPlaying)
-
-      useBlobURL()
-    }
-
-    function createElem (time) {
-      if (!elem) {
-        elem = document.createElement(tagName)
-        elem.controls = true
-        elem.autoplay = true // for chrome
-        elem.play() // for firefox
-
-        elem.addEventListener('progress', function () {
-          currentTime = elem.currentTime
-        })
-
-        rootElem.appendChild(elem)
-      }
+  render(file, function (tagName) {
+    if (tagName === 'video' || tagName === 'audio') return createMedia(tagName)
+    else return createElem(tagName)
+  }, function (err, elem) {
+    if (err && elem) {
+      elem.remove()
     }
-  }
 
-  function onPlaying () {
-    elem.removeEventListener('playing', onPlaying)
-    cb(null, elem)
-  }
+    cb(err, elem)
+  })
 
-  function appendToAudio () {
-    elem = document.createElement('audio')
+  function createMedia (tagName) {
+    var elem = createElem(tagName)
     elem.controls = true
-    elem.autoplay = true
-    rootElem.appendChild(elem)
-    file.getBlobURL(function (err, url) {
-      if (err) return fatalError(err)
-      elem.addEventListener('error', fatalError)
-      elem.addEventListener('playing', onPlaying)
-      elem.src = url
-      elem.play()
-    })
+    elem.autoplay = true // for chrome
+    elem.play() // for firefox
```

Comment:
`createMedia` needs to append to the DOM too. Right now, only `createElem` does.